src/main/java/edu/rice/pcdp/runtime/Runtime.java

```diff
@@ -54,6 +55,14 @@ public static void resizeWorkerThreads(final int numWorkers)
         SystemProperty.numWorkers.set(numWorkers);
         taskPool = new ForkJoinPool(numWorkers);
     }
+    
+    public static void setExecutorService( ExecutorService executorService ) throws InterruptedException {
```

Comment:
In addition to just the standard documentation, it might even be helpful to add a note on common use cases of this capability (which for now would just be your use of cached thread pool) and explanations of why these other executor services might be preferrable.